LibreNMS/Util/Url.php

```diff
@@ -139,7 +139,7 @@ public static function portLink($port, $text = null, $type = null, $overlib = tr
             $text = $label;
         }
 
-        $content = '<div class=list-large>' . addslashes(htmlentities($port->device->displayName() . ' - ' . $label)) . '</div>';
+        $content = '<div class=list-large>' . addslashes(htmlentities((isset($port->device) ? $port->device->displayName() : '') . ' - ' . $label)) . '</div>';
```

Comment:
you should split this to its own PR

Reply:
Additionally, calling portLink on the port of a deleted device is not going to be useful

Reply:
Ok, let's solidify this code and all fixes, and I'll separate this into two new PRs.